Services/Simulations.cs

```diff
@@ -50,6 +54,8 @@ public interface ISimulations
         // Change the simulation, setting the device deletion complete
         Task<bool> TryToSetDeviceDeletionCompleteAsync(string simulationId);
 
+        Task<bool> TryToUpdateSimulation(Models.Simulation simulation);
```

Comment:
I don't think this is required, you can use `UpsertAsync`

Reply:
thanks, updated.